Style fixes

bashunaimiroy opened the pull request:
Hello folks!

**Description**
This PR has a parcel of assorted style changes. Some themes are: Changing classnames to be more [BEM-compliant](http://getbem.com/introduction/), refactoring styles to DRY, and addressing homepage layout issues.

**Testing steps:**

1. Navigate to the homepage
2. Resize the screen width between 800-1200px. Ensure the text fills the screen pleasingly
3. Resize screen to 320px (iPhone 5)
4. Ensure the Twitter handle in the "Help" box in the homepage introduction doesn't overflow its container.

tatianamac commented:
Thank you, @bashunaimiroy , for your first PR! Before I dig into it, @ovlb , do you have time/desire/energy to review this? I'd like to move forward with implementing BEM but ofc would love your thoughts since you architected much of this :)

ovlb commented on 11ty/index.njk:
```diff
@@ -3,22 +3,22 @@ layout: layouts/index.njk
+  </div>
+  <div class="box">
+    <h2 class="subtitle">Ways to help</h2>
+    <ol class="box__list">
```
This list is a structure that can stand on its own. Maybe we shouldn’t use an element class here, but a new block. Agreed, though, that `help` wasn’t the most useful name as it is too tightly coupled to the content.

ovlb commented on assets/css/structures/_header.scss:
```diff
@@ -1,4 +1,4 @@
-.title__thicc {
+.title--thicc {
```
These styles are doubled. They exist in the typography partial as well as here. I think this is the more up-to-date version. Can you move these into typography and delete them here.

Maybe we should also make the naming of headlines more consistent. Currently, we have `title--thicc`, `main-headline` and `subtitle`. I would propose `thicc-headline`, `main-headline`, and `sub-headline`.

ovlb commented on assets/css/structures/_header.scss:
```diff
@@ -19,17 +19,17 @@
   }
 
-  .title__thicc + p {
+  .title--thicc + p {
```
These styles make the class hard to reuse. Maybe add a class `inctroduction` to the homepage and scope these to `introduction__title` and `inctroduction__intro` or something similar.

bashunaimiroy replied:
heartily agree. I will have bandwidth this weekend, I'll take a moment to fix them all

bashunaimiroy replied:
thanks for the thorough review ovlb!

ovlb replied:
Great, thanks :)

bashunaimiroy commented:
hi @ovlb, I've taken your good code review to heart and refactored quite a lot. Prominent bits:

- I've refactored the whole header section to separate grid layout styles and typography. 
- It's also now a `<header>` element
- moved all typography from `header.scss` to `typography.scss`. My assumption is that `header.scss` should deal with header layouts, like the new `.introduction` class
- I also took a couple of small design liberties to make it more readable on small screens. the `.box` element and some headings were squished and/or visually buggy on 320px and 375px screens. I am hoping this will fix that, since people will probably want to refer to it on mobile. 

Thanks!

ovlb commented:
@bashunaimiroy Thanks! That’s looking good so far. 

Adding a `header` element was a very good idea. I am only wondering if it makes more sense to change the whole introduction `section` to be the `header` element, instead of having only the `h1` in it.

Because [as the spec says](https://www.w3.org/TR/html52/sections.html#the-header-element):

>  A header typically contains a group of introductory or navigational aids.

bashunaimiroy commented:
@ovlb I was wondering same thing. It's would fit a <header> more than a <section>, which is currently is.

ovlb commented:
@bashunaimiroy You should be able to remove the wrapper and place the `h1` directly on the grid using the `introduction__title` class.

tatianamac commented:
@ovlb Please feel free to approve this PR once you feel good about it.

Thank you, @bashunaimiroy ! Once this is merged in, would it be okay if I thanked you for your contribution on Self-Defined's Twitter?

bashunaimiroy commented:
@ovlb es ist vollbracht

bashunaimiroy commented:
@tatianamac yes, that would be lovely, thank you! I'll go clean up my twitter bio
